docs/handbook/systems-administration.md

```diff
@@ -751,6 +751,23 @@ onto the system before changing the run-level.
    process is not interrupted by system prompts requiring user-interactive
    intervention.
 
+
+## Updating All Packages (Upgrading Your System)
+
+OpenIndiana Hipster is a rolling release
+distribution, so to upgrade your system, you simply run `pkg update`
+all packages will be updated to the latest revision:
+
+```sh
+sudo pkg update
+```
```

Comment:
The canonical way of doing this on Solarish systems is
pfexec pkg update

Reply:
I was wondering about that. I'll try to do a future update and maybe tag you as a reviewer, and add a section on the differences between the two (without going into a lot of detail on `pfexec` - just sort of a landing entry for people more familiar with `sudo` - I'm quite rusty on `pfexec`, and wasn't sure if things had changed).

Reply:
So one existing stumbling block, which is interesting:



```$ pfexec pkg update

pkg update: Insufficient access to complete the requested operation.
Please try the operation again as a privileged user.
```

Somehow sudo seems to bypass the privilege mechanisms pfexec is using.

Reply:
Solaris and its derivates have fine grain rights and privilege control, look at my user entry in /etc/user_attr:
andreas::::type=normal;defaultpriv=basic,dtrace_user,file_dac_read,file_dac_write,proc_clock_highres;profiles=Software Installation,CPU Power Management,Service Management,ZFS File System Management

In this case you'll need the profiles=Software Installation part to be able to use pfexec pkg.

Reply:
Thanks - my last memory of using OI or Solaris, almost a decade ago, was that `sudo` didn't even exist in the installation of the OS, though maybe I'm remembering incorrectly. Was a decision made to include it for compatibility?

Reply:
You can configure sudo to also don't need a password. You have even configure it to allow passwordless for limited paths. The mechanisms in Solaris evolved over several versions. IIRC roles have been introduced in Solaris 8 while privileges came with Solaris 10.

Reply:
You might also want to add that one need to have the Software Installation profile in /etc/user_attr for this to work.

Reply:
Sounds good - I added a variant that doesn't require the user to manually edit `/etc/user_attr`